agent/consul/server.go

```diff
@@ -601,9 +601,9 @@ func (s *Server) setupRaft() error {
 	}
 
 	// Set up a channel for reliable leader notifications.
-	leaderCh := make(chan bool, 1)
-	s.config.RaftConfig.NotifyCh = leaderCh
-	s.leaderCh = leaderCh
+	raftReconcileChannel := make(chan bool, 1)
```

Comment:
The name "reconcile" has a special meaning where we sync the leader's Serf members with the catalog, so I'd avoid that and change `raftReconcileCh` to `raftNotifyCh` for the local variable here, or you could do:

```
s.raftNotifyCh = make(chan bool, 1)
s.config.RaftConfig.NotifyCh = s.raftNotifyCh
```